src/ratelimit_action.rs

```diff
@@ -141,11 +142,40 @@ impl RateLimitAction {
             entries.extend(conditional_data.entries()?);
         }
 
+        let known_attributes: Vec<String> = vec![
+            "ratelimit.domain".to_string(),
+            "ratelimit.hits_addend".to_string(),
```

Comment:
Can we consolidate that vec? We're building it twice (again in `get_known_attributes`)... mostly concerned we don't keep them in sync